docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,130 @@
+# Syncing LDAP Groups to OpenShift Groups
+
+OpenShift cluster admins need a way to sync their OpenShift `Groups` to external records in an LDAP server. This requires that we have a way of retrieving the LDAP users that are the members of a group, determining the OpenShift `Group` name for this collection, and updating the OpenShift record, removing the record as necessary when the corresponding LDAP record is removed. Necessary information about the LDAP server set-up and schema will be provided by the user in a `yaml` configuration file and referenced in the `oadm sync-groups` command.
+
+## Use Cases
+1. I want to sync all LDAP groups minus some blacklist to OpenShift `Groups`
+* I want to sync a specific set of LDAP groups to OpenShift `Groups`
+* I want to sync all OpenShift groups minus some blacklist with the LDAP server
```

Comment:
> That's not true - it's not in this proposal for succinctness, but all filters are going to be generated with a simple mapping of OpenShift Group fields from ObjectMeta to the filter. For instance, lets say:
> 
> Our group has Name mygroup and an Annotation like openshift.io/someAnnotation=foobar.
> filterBase = (&(%s)(%s))
> mapping["DN"] = "Name"
> mapping["someOtherAttribute"] = "Annotations.someAnnotation
> The result would be that DN=mygroup would fill the the first %s and someOtherAttribute=foobar would fill the second.

I don't remember discussing that.  Ordering through a map doesn't work.  It could be done with a golang template, but no one would get that right.  It might be useful to allow additional annotations to be created, but I wouldn't have expected them to be used anywhere automatically.

In either case, it's not pertinent to this particular question.  We must have the expected annotations on the openshift group in order to make the reverse mapping.